### DISCUSSION TITLE
install problems

### DISCUSSION THREAD

**Discussion by ibaboomer** [danielmiessler/Fabric#147] [Q&A]
Having an issue getting it running:

fabric % curl -sSL https://install.python-poetry.org | python3 -
Retrieving Poetry metadata

# Welcome to Poetry!

This will download and install the latest version of Poetry,
a dependency and package manager for Python.

It will add the `poetry` command to Poetry's bin directory, located at:

/Users/myg/.local/bin

You can uninstall at any time by executing this script with the --uninstall option,
and these changes will be reverted.

Installing Poetry (1.8.1): Creating environment
Traceback (most recent call last):
  File "<stdin>", line 945, in <module>
  File "<stdin>", line 923, in main
  File "<stdin>", line 560, in run
  File "<stdin>", line 581, in install
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/contextlib.py", line 117, in __enter__
    return next(self.gen)
  File "<stdin>", line 653, in make_env
  File "<stdin>", line 639, in make_env
  File "<stdin>", line 317, in make
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/venv/__init__.py", line 66, in __init__
    self.symlinks = should_use_symlinks(symlinks)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/venv/__init__.py", line 31, in should_use_symlinks
    raise Exception("This build of python cannot create venvs without using symlinks")
Exception: This build of python cannot create venvs without using symlinks

fabric % ./setup.sh
Installing python dependencies
./setup.sh: line 12: poetry: command not found
Updating /Users/myg/.bashrc
Command fabric not found in the current Poetry environment.
Command fabric-api not found in the current Poetry environment.
Command fabric-webui not found in the current Poetry environment.
Updating /Users/arberg/.zshrc
Command fabric not found in the current Poetry environment.
Command fabric-api not found in the current Poetry environment.
Command fabric-webui not found in the current Poetry environment.
Updating /Users/myg/.bash_profile
Command fabric not found in the current Poetry environment.
Command fabric-api not found in the current Poetry environment.
Command fabric-webui not found in the current Poetry environment.
To apply the changes, please run the following command(s) in your terminal:
source /Users/myg/.bashrc
source /Users/myg/.zshrc
source /Users/myg/.bash_profile
fabric 

fabric % fabric --setup
zsh: command not found: fabric

**Comment by ibaboomer** [comment]
I do not see where it added anything to the profile

**Reply by ibaboomer** [reply]
and the answer is:    curl -sSL https://install.python-poetry.org | sed 's/symlinks=False/symlinks=True/' | python3 -

**Reply by headbug** [reply]
Or even better, if you have the possibility, is to use a newer version of python. 3.9 is rather old now, ca 3 years.

**Reply by ibaboomer** [reply]
To be clear this issue a brand new OSX which I installed Homebrew, Xcode and Vscode with Python.   That is the only thing installed so far.

**Comment by lacalasolutions** [comment]
hello I followed all the install instructions on my macOS, and when I try to run fabric I get the following 
"error Failed to fetch models: HTTP 401"
<img width="367" alt="Screenshot 2024-03-11 at 3 44 54 PM" src="https://github.com/danielmiessler/fabric/assets/122548923/d1407eb1-0cc3-40e1-868d-d14e0139f042">
I've check through all the documentation and discussions, and haven't seen a mention of it. Any help would be greatly appreciated.
-Eric

**Reply by xssdoctor** [reply]
Your api key is probably not correctly entered. Look in ~/.config/fabric/.env

**Comment by lacalasolutions** [comment]
Thank you!

**Comment by lacalasolutions** [comment]
Ok , I checked, they are the correct api keys, I also have a lot of unused calls.
